src/Viewer.tsx

```diff
@@ -21,30 +22,69 @@ export interface WidgetModelWithState extends WidgetModel {
 
 export interface WidgetProps {
   controller: SimulariumController;
-  height: number;
-  width: number;
 }
 
 function ViewerWidget(props: WidgetProps): JSX.Element {
   const [modelInfo, setModelInfo] = useState<ModelInfo | undefined>({});
   const [trajectoryTitle, setTrajectoryTitle] = useState<string | undefined>(
     ''
   );
+  const [dimensions, setDimensions] = useState({ width: 500, height: 529 });
+  const [showSidePanel, setShowSidePanel] = useState(true);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const observerRef = useRef<ResizeObserver | null>(null);
+
+  useEffect(() => {
+    // Initialize ResizeObserver if it doesn't exist
+    if (!observerRef.current && containerRef.current) {
+      observerRef.current = new ResizeObserver(
+        (entries: ResizeObserverEntry[]) => {
+          for (const entry of entries) {
+            // get the size of viewer container
+            let { width } = entry.contentRect;
+            const { height } = entry.contentRect;
+            // hide side panel if space is small
+            setShowSidePanel(width > 580);
+            if (showSidePanel) {
+              width = width - 280;
+            }
```

Comment:
Just a note, I think this is also possible with the [`@media` selector in CSS](https://github.com/allen-cell-animated/nucmorph-colorizer/blob/main/src/App.module.css#L138)! I like this approach though, because ResizeObserver makes the behavior a bit more explicit.

Reply:
Would be curious to think through the pros and cons of that, especially if mediua queries could reduce flickering, maybe for a future issue/refactor?